components/TextFields/src/MDCTextField.m

```diff
@@ -0,0 +1,456 @@
+// textContainerInsets value the positioning delegate can return to inset this text rect.
+- (CGRect)textRectForBounds:(CGRect)bounds {
+  CGRect textRect = bounds;
+
+  // Standard textRect calculation
+  UIEdgeInsets textContainerInset = [_coordinator textContainerInset];
+  if (self.mdc_effectiveUserInterfaceLayoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
+    textRect.origin.x += textContainerInset.right;
+  } else {
+    textRect.origin.x += textContainerInset.left;
+  }
+  textRect.size.width -= textContainerInset.left + textContainerInset.right;
+
+  // Adjustments for .leftView, .rightView
+  CGFloat leftViewWidth = CGRectGetWidth([self leftViewRectForBounds:bounds]);
+  CGFloat rightViewWidth = CGRectGetWidth([self rightViewRectForBounds:bounds]);
+  rightViewWidth += MDCTextInputTextRectRightViewClearPaddingCorrection;
+
+  if (self.leftView.superview) {
+    textRect.size.width -= leftViewWidth;
+  }
+  if (self.rightView.superview) {
+    textRect.size.width -= rightViewWidth;
+  }
+
+  // Adjustments for RTL and clear button
+  // .leftView and .rightView actually are leading and trailing: their placement is reversed in RTL.
+  CGFloat clearButtonWidth = CGRectGetWidth(self.clearButton.bounds);
+  clearButtonWidth += MDCTextInputTextRectRightViewClearPaddingCorrection;
+
+  if (self.mdc_effectiveUserInterfaceLayoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
+    // EITHER the rightView or clear button will be shown.
+    if (self.rightView.superview) {
+      textRect.origin.x += rightViewWidth;
+    } else {
+      // Clear buttons are only shown if there is entered or set text to clear.
+      if (self.text.length > 0) {
+        switch (self.clearButtonMode) {
+          case UITextFieldViewModeAlways:
+          case UITextFieldViewModeUnlessEditing:
+            textRect.size.width -= clearButtonWidth;
+            textRect.origin.x += clearButtonWidth;
+            break;
+          default:
+            break;
+        }
+      }
+    }
+  } else {
+    if (self.leftView.superview) {
+      textRect.origin.x += leftViewWidth;
+    }
+    if (self.rightView.superview) {
+    } else {
+      if (self.text.length > 0) {
+        switch (self.clearButtonMode) {
+          case UITextFieldViewModeAlways:
+          case UITextFieldViewModeUnlessEditing:
+            textRect.size.width -= clearButtonWidth;
+            break;
+          default:
+            break;
+        }
+      }
+    }
+  }
+
+  // UITextFields have a centerY based layout. And you can change EITHER the height or the Y. Not
+  // both. Don't know why. So, we have to leave the text rect as big as the bounds and move it to a
+  // Y that works.
+  CGFloat actualY =
+      (CGRectGetHeight(bounds) / 2.f) - MDCRound(MAX(self.font.lineHeight,
+                                                     self.placeholderLabel.font.lineHeight) /
+                                                 2.f);  // Text field or placeholder
```

Comment:
This is the way clang-format has done it. Should I try to override it? No matter what I do it will put it back.